### PR TITLE
User Tests

### PR DESCRIPTION
**Pull request by phump81**
Feature tests implemented to test:

User sign up:

- Successful sign up
- Invalid email address
- Password too short
- Password too long
- No first name
- Passwords do not match

User login:

- Successful login
- Invalid email
- Unrecognised email
- Incorrect password

User Logout:

- Successful logout

Added extra method to custom_helper.rb.

All tests passing.

### CODE REVIEW

**Comment by melanieanderson1995** [comment]
Looksgood to me, thanks paul for finishing the work we started on friday bigups

File: spec/features/user_login_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.feature "Login", type: :feature do
+  scenario "successful login" do
+    create_user
+    expect(page).to have_content("Logged in as: test")
+    log_out_and_in
+    fill_in "session[email]", with: "test@test.com"
+    fill_in "session[password]", with: "123123"
+    click_button "Log in"
```
Comment (02ship) [inline spec/features/user_login_spec.rb:10]: Could these three lines also be added to log_out_and_in? The naming of the method implies that login would be completed

Comment (phump81) [reply spec/features/user_login_spec.rb:10]: I left them out, so that I could use incorrect login information on the tests.

Comment (dwram) [reply spec/features/user_login_spec.rb:10]: Other than this comment, i absolutely agree. Once this change is made, I think this is the perfect PR. Great description, tests deduce behaviour and state of the process.

**Review by 02ship — APPROVED** [review]

**Comment by ralphm10** [comment]
Reviewed and approved

**Comment by dwram** [comment]
Happy to approve. Changes requested were purely subjective. Great PR.

**Comment by ROCHAAL** [comment]
Reviewed and approved

**Pull request merged**